test/e2e/node_reuse.go

```diff
@@ -204,26 +240,32 @@ func nodeReuse(ctx context.Context, inputGetter func() NodeReuseInput) {
 		time.Sleep(30 * time.Second)
 	}
 
-	By("Untaint all CP nodes")
-	// The rest of CP nodes may take time to be untaintable
-	// We have untainted the 2 first CPs
-	for untaintedNodeCount := 0; untaintedNodeCount < numberOfControlplane-2; {
-		controlplaneNodes = getControlplaneNodes(ctx, clientSet)
-		untaintedNodeCount = untaintNodes(ctx, targetClusterClient, controlplaneNodes, controlplaneTaints)
-		time.Sleep(10 * time.Second)
-	}
-
-	By("Scale the controlplane down to 1")
+	By("Scale the controlplane down to 1 [node_reuse]")
 	ScaleKubeadmControlPlane(ctx, managementClusterClient, client.ObjectKey{Namespace: input.Namespace, Name: input.ClusterName}, 1)
 
-	Byf("Wait until controlplane is scaled down and %d BMHs are Available", numberOfControlplane)
+	Byf("Wait until controlplane is scaled down and %d BMHs are Available [node_reuse]", (numberOfControlplane - 1 + numberOfWorkers))
 	WaitForNumBmhInState(ctx, bmov1alpha1.StateAvailable, WaitForNumInput{
 		Client:    managementClusterClient,
 		Options:   []client.ListOption{client.InNamespace(input.Namespace)},
 		Replicas:  numberOfControlplane,
```

Comment:
`Replicas` should match the expression in the message. Otherwise this can get very confusing.